### DISCUSSION TITLE
Compiler error in hb-paint.h on Solaris

### DISCUSSION THREAD

**Discussion by dmarkov20** [harfbuzz/harfbuzz#4188] [General] [answered]
I get the following compilation error when trying to build JDK 11u with HarfBuzz version 7.0.1 on Solaris platform:

```
libharfbuzz/hb-paint.h, line 620: error: trailing "," prohibited in enum declaration (E_TRAILING_COMMA_IN_ENUM)
CC: Warning: Option -fsingle passed to ld, if ld is invoked, ignored otherwise
libharfbuzz/hb-paint.h, line 620: Error: Identifier expected instead of "}".
 1 Error(s) detected.
```

The root cause of the error is that `hb-paint.h` has the enum with trailing comma. As far as I know many C++ compilers support the following syntax:` enum {FOO,}`. However it is not the case for Solaris Studio 12u4 and some other  compilers. Is it possible remove the comma for better compatibility?

**Comment by behdad** [comment, marked as answer]
Fixed. Thanks.
